rq/job.py

```diff
@@ -733,4 +732,43 @@ def register_dependency(self, pipeline=None):
             connection.sadd(dependents_key, self.id)
             connection.sadd(self.dependencies_key, dependency_id)
 
+    def get_dependencies_statuses(
```

Comment:
I'm not sure whether we should have this API, or we should just expose a `job.has_unmet_dependencies` that simply returns a boolean indicating whether or not all dependencies are finished.

Reply:
@selwin I've changed such that `Job` now has a method `dependencies_finished`. I'm not in love with the name, but including a _negative_ seemed to make a few things more convoluted; e.g:
```
[ ... for ... in 
if not job.has_unmet_dependencies]
```

Originally I thought exposing `get_dependencies_statuses` might be of use to other projects,(like `django_rq`) but I think you're suggestion is better, because it keeps the loop in `Queue#enqueue_dependents` cleaner.